### ISSUE TITLE
Full configuration leads to server side memory errors

### ISSUE THREAD

**Issue by sbutcher-arm** [ARMmbed/mbedtls#2069]
### Description
- Type: Bug
- Priority: Major

---------------------------------------------------------------
## Bug

**OS**  
Mbed OS|linux|windows|

**mbed TLS build:**  
Version: Commit 0592ea772aee

With a full configuration, `compat.sh` fails with multiple memory errors. To reproduce:

1. `./scripts/config.pl full`
2. `make clean; make`
3. `cd tests; ./compat.sh --memcheck`

The result is a total of 78 server memory errors!

**Comment by hanno-arm** [comment]
First test reporting memory errors for me is 
```
./tests/compat.sh --memcheck --filter "ECDH-ECDSA-AES256-SHA" -p OpenSSL -m tls1 -V no -v
```

**Comment by hanno-arm** [comment]
@sbutcher-arm This fails for 2.7 and 2.1, too, so it looks like an old problem going unnoticed so far.

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-2569

**Comment by hanno-arm** [comment]
Fixed by #2070.
